<?xml version="1.0"?>
<launch>

    <arg name="host"/>
    <arg name="max_velocity" default="5"/>
    <arg name="heading_update_rate" default="10"/> <!-- Update rate in seconds -->
    <arg name="motor" default="true"/>
    <arg name="max_avoid_threshold" default="3"/>
    <arg name="min_avoid_threshold" default="2"/>
    <arg name="people_list"/>
    <arg name="offset_x"/>
    <arg name="offset_y"/>
    <arg name="trace_rgba"/>
    <arg name="waypoint_file"/>

    <rosparam file="$(find multi_drone_sar)/data/$(arg people_list).yaml" command="load"/>
    
    
    <param name="offset_x" value="$(arg offset_x)"/>
    <param name="offset_y" value="$(arg offset_y)"/>
    <param name="trace_rgba" type="yaml" value="$(arg trace_rgba)"/>

    <node name="fake_drone" pkg="multi_drone_sar" type="fake_drone.py" output="screen" respawn="false" respawn_delay="0">
        <param name="host" value="$(arg host)"/>
        <param name="max_charge_distance" value="6"/>
    </node>

    <node name="feelforce" pkg="multi_drone_sar" type="feelforce.py" output="screen"/>
    <node name="runaway" pkg="multi_drone_sar" type="runaway.py" output="screen">
        <param name="threshold" value="$(arg min_avoid_threshold)"/>
    </node>
    <group if="$(arg motor)">
        <node name="motor" pkg="multi_drone_sar" type="motor.py" output="screen">
            <param name="host" value="$(arg host)"/>
            <remap from="runaway/heading" to="runaway_suppressor/heading"/>
        </node>
    </group>
    <!--<node name="wander" pkg="multi_drone_sar" type="wander.py" output="screen">
        <param name="velocity" value="$(arg max_velocity)"/>
        <param name="heading_update_rate" value="$(arg heading_update_rate)"/>
    </node>-->

    <node name="avoid" pkg="multi_drone_sar" type="avoid.py" output="screen">
        <param name="max_threshold" value="$(arg max_avoid_threshold)"/>
        <param name="min_threshold" value="$(arg min_avoid_threshold)"/>
        <remap from="wander/heading" to="wander_suppressor/heading"/>
    </node>

    <node name="navigation" pkg="multi_drone_sar" type="navigation.py" output="screen" respawn="false" respawn_delay="0">
        <rosparam file="$(find multi_drone_sar)/data/$(arg waypoint_file).yaml" command="load"/>
        <param name="offset_x" value="$(arg offset_x)"/>
        <param name="offset_y" value="$(arg offset_y)"/>
    </node>

    <node name="neighbour_listener" pkg="multi_drone_sar" type="neighbour_listener.py" output="screen"></node>
    <node name="formation" pkg="multi_drone_sar" type="formation.py" output="screen" respawn="false" respawn_delay="0"></node>

    <node name="find_people" pkg="multi_drone_sar" type="find_person.py" output="screen" respawn="false" respawn_delay="0">
        <param name="proximity_threshold" value="30"/>
    </node>
    <node name="multiple_people" pkg="multi_drone_sar" type="multiple_people.py" output="screen" respawn="false" respawn_delay="0">
        <param name="time_threshold" value="3"/>
    </node>
    <node name="spiral" pkg="multi_drone_sar" type="spiral.py" output="screen" respawn="false" respawn_delay="0">
        <param name="spiral_size" value="200"/>
        <param name="plot_spiral" value="true"/>
    </node>

    <!-- SUPPRESSORS/INHIBITORS -->
    <node name="wander_suppressor" pkg="multi_drone_sar" type="suppressor.py"  output="screen">
        <param name="time_suppressed" value="25"/>
        <param name="upper_topic" value="navigation_suppressor/heading"/>
        <param name="lower_topic" value="navigation_suppressor/heading"/>
        <remap from="~output" to="wander_suppressor/heading"/>
    </node>
    <node name="runaway_suppressor" pkg="multi_drone_sar" type="suppressor.py"  output="screen">
        <param name="time_suppressed" value="25"/>
        <param name="upper_topic" value="avoid/heading"/>
        <param name="lower_topic" value="runaway/heading"/>
        <remap from="~output" to="runaway_suppressor/heading"/>
    </node>
    <node name="navigation_suppressor" pkg="multi_drone_sar" type="suppressor.py"  output="screen">
        <param name="time_suppressed" value="25"/>
        <param name="upper_topic" value="formation_suppressor/heading"/>
        <param name="lower_topic" value="navigation/heading"/>
        <remap from="~output" to="navigation_suppressor/heading"/>
    </node>
    <node name="formation_suppressor" pkg="multi_drone_sar" type="suppressor.py"  output="screen">
        <param name="time_suppressed" value="70"/>
        <param name="upper_topic" value="spiral/heading"/>
        <param name="lower_topic" value="formation/heading"/>
        <remap from="~output" to="formation_suppressor/heading"/>
    </node>

</launch>